CMakeLists.txt

```diff
@@ -14,8 +14,8 @@ IF(CMAKE_BUILD_TYPE MATCHES Debug)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_DEBUG} -Wall -g -O0 -fprofile-arcs -ftest-coverage")
 ENDIF(CMAKE_BUILD_TYPE MATCHES Debug)
 
-set(Boost_USE_MULTITHREADED OFF)
-find_package(Boost COMPONENTS unit_test_framework REQUIRED)
+set(Boost_USE_MULTITHREADED ON)
```

Comment:
Use a parameter for this. Lets keep it OFF by default until we have all working. Also, why we need Boost to be multithread?